components/SplitPane/common/src/commonMain/kotlin/org/jetbrains/compose/splitpane/SplitPaneDSL.kt

```diff
@@ -0,0 +1,176 @@
+package org.jetbrains.compose.splitpane
+
+import androidx.compose.foundation.InteractionState
+import androidx.compose.foundation.gestures.detectDragGestures
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.input.pointer.consumeAllChanges
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+
+/** Receiver scope which is used by [HorizontalSplitPane] and [VerticalSplitPane] */
+interface SplitPaneScope {
+
+    /**
+     * Set up first composable item if SplitPane, for [HorizontalSplitPane] it will be
+     * Right part, for [VerticalSplitPane] it will be Top part
+     * @param minSize a minimal size of composable item.
+     * For [HorizontalSplitPane] it will be minimal width, for [VerticalSplitPane] it wil be minimal Heights.
+     * In this context minimal mean that this composable item could not be smaller than specified value.
+     * @param content composable item content.
+     * */
+    fun first(
+        minSize: Dp = 0.dp,
+        content: @Composable () -> Unit
+    )
+
+    /**
+     * Set up second composable item if SplitPane.
+     * For [HorizontalSplitPane] it will be Left part, for [VerticalSplitPane] it will be Bottom part
+     * @param minSize a minimal size of composable item.
+     * For [HorizontalSplitPane] it will be minimal width, for [VerticalSplitPane] it wil be minimal Heights.
+     * In this context minimal mean that this composable item could not be smaller than specified value.
+     * @param content composable item content.
+     * */
+    fun second(
+        minSize: Dp = 0.dp,
+        content: @Composable () -> Unit
+    )
+
+    /**
+     * Set up splitter that will split first and second parts of split pane.
+     * By default for [HorizontalSplitPane] it will be vertical line that divides available width in some place.
+     * For [VerticalSplitPane] it wil be horizontal line that divides available heights in some place.
+     * @param block specify configuration lambda for [SplitterScope] receiver
+     * which wil allow to configure custom composable for splitter and move handle
+     * */
+    fun splitter(
+        block: SplitterScope.() -> Unit
+    )
+}
+
+/** Receiver scope which is used by [HorizontalSplitPane] and [VerticalSplitPane] by [SplitPaneScope] */
+interface SplitterScope {

```

Comment:
1. Not sure we need `isHorizontal` here. Clients of the library already know if it isHorizontal:
```
VerticalSplitPane {
    splitter {
        VerticalSplitter()
    }
}

HorizontalSplitPane {
    splitter {
        HorizontalSplitter()
    }
}
```

2. For creating custom splitter we still need access to `state` (to check if state.moveEnabled):
```
fun SplitterScope.CustomVerticalSplitter(state: SplitPaneState)
```

3. Technically speaking, we don't need SplitterScope to make custom splitter. All we need is only `state`:
```
fun CustomVerticalSplitter(state: SplitPaneState)
```
But how can we write it without copy-paste code?

Probably all my reasoning is very subjective. So the best decision would be to just remove `SplitterScope` and make `markAsHandle` internal, until we figure out how we can implement a conscience API for creating custom splitters (we should investigate a lot of user cases). Adding things is easier than removing/changing them :)

Reply:
Well. Possibly done.

I remove old splitter DSL first. But then think a little and try another approach.
Now we could specify visible part and handle part separately.
That allow to customize handle, and measure split pane parts and splitter, and place them so splitter content wouldn't overlap split pane parts.
Also we could set up handle position according to splitter, Before, Above or After.
Also movementEnable check now moved to layout, so we doesnt place handle if movement not enabled.

Possibly this solution is better than the previous one but, if you still want to remove splitter customisation DSL. I think it wouldn't be a problem, at least at that moment.

Reply:
This is much better :)